File: ApiBaseCore/ApiBaseCore.Application/DTOs/RegistroUsuarioDto.cs
namespace ApiBaseCore.Application.DTOs;

/// <summary>
/// Define los datos que un cliente debe enviar para registrarse.
/// Las reglas de validación para esta clase se definen en RegistroUsuarioDto.cs
/// </summary>
public class RegistroUsuarioDto
{
    public required string Name { get; set; }
    public required string Email { get; set; }
    public required string Password { get; set; }
    public required string Phone { get; set; }
    public string? TaxId { get; set; }
}

File: ApiBaseCoreJava/src/main/java/com/polancou/apibasecore/application/dtos/RegistroUsuarioDto.java
package com.polancou.apibasecore.application.dtos;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;

@Data
public class RegistroUsuarioDto {
    @NotBlank
    private String name;
    
    @NotBlank
    @Email
    private String email;
    
    @NotBlank
    private String password;
    
    @NotBlank
    private String phone;
    
    private String taxId;
}
